src/Sarif.Viewer.VisualStudio/SarifSnapshot.cs

```diff
@@ -0,0 +1,289 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Windows.Forms;
+
+using EnvDTE;
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Microsoft.CodeAnalysis.Sarif.Driver.Sdk;
+using Microsoft.CodeAnalysis.Sarif.Sdk;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell.TableManager;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace SarifViewer
+{
+    class SarifSnapshot : TableEntriesSnapshotBase
+    {
+        private string _projectName;
+        private readonly List<SarifError> _errors;
+        private Dictionary<string, string> _remappedFilePaths;
+        private List<Tuple<string, string>> _remappedPathPrefixes;
+        private Dictionary<string, NewLineIndex> _fileToNewLineIndexMap;
+
+        internal SarifSnapshot(string filePath, IEnumerable<SarifError> errors)
+        {
+            FilePath = filePath;
+            _errors = new List<SarifError>();
+            _errors.AddRange(errors);
+
+            _remappedFilePaths = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            _remappedPathPrefixes = new List<Tuple<string, string>>();
+
+            _fileToNewLineIndexMap = new Dictionary<string, NewLineIndex>(StringComparer.OrdinalIgnoreCase);
+        }
+
+        public override int Count
+        {
+            get { return _errors.Count; }
+        }
+
+        public IList<SarifError> Errors
+        {
+            get { return _errors; }
+        }
+
+        public string FilePath { get; }
+
+        public override int VersionNumber { get; } = 1;
+
+        public override bool TryGetValue(int index, string columnName, out object content)
+        {
+            content = null;
+
+            if ((index >= 0) && (index < _errors.Count))
+            {
+                if (columnName == StandardTableKeyNames.DocumentName)
+                {
+                    content = FilePath;
+                }
+                else if (columnName == StandardTableKeyNames.ErrorCategory)
+                {
+                    content = Constants.VSIX_NAME;
```

Comment:
Didn't you say something at some point about using the `Category` item in the `result` object property bag, if there was one?